### PR TITLE
Update help.py

### PR DESCRIPTION
**Pull request by AakashS-chan**
Updated the whole button help interaction command.

### CODE REVIEW

**Review by rf20008 — APPROVED** [review]
LGTM assuming you've tested (a confirm button is a good idea :))

**Comment by AakashS-chan** [comment]
Tested in 2 different guilds.

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: cogs/help.py [hunk 1]
```diff
@@ -3,28 +3,55 @@
...
+        view = Confirmation()
+        await interaction.response.send_message('Click "Yes!" to create the thread.', view=view, ephemeral=True)
+        global help_topic
+        help_topic = 'Python'
+        
+        
+class Confirmation(nextcord.ui.View):        
+        
+    @nextcord.ui.button(label='Yes!', style=nextcord.ButtonStyle.success)
+    async def Confirm(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
```
Comment (sairam4123) [inline cogs/help.py:38]: Function names should snake_case

File: cogs/help.py [hunk 2]
```diff
@@ -3,28 +3,55 @@
...
+        await interaction.response.send_message('Created the thread', ephemeral=True)
+        thread = await interaction.channel.create_thread(name=f" {help_topic} help ({interaction.user})", type=ChannelType.public_thread)
         await thread.add_user(interaction.user)
+        await thread.send(f'<@&882192899519954944> Help needed!\nAlright now that we are all here to help, what do you need help with {interaction.user.mention}?')
```
Comment (sairam4123) [inline cogs/help.py:42]: It might be better if the mention is not hardcoded in the message.

File: cogs/help.py [hunk 3]
```diff
@@ -3,28 +3,55 @@
...
-        await log_channel.send(f"Help thread for {name} created by {interaction.user.mention}: {thread.mention}!")
-        await thread.send(f"<@&882192899519954944> Help needed!\nAlright now that we are all here to help, what do you need help with {interaction.user.mention}?")
+        await log_channel.send(f"{help_topic} help thread created by {interaction.user.mention}: {thread.mention}!")
+        self.stop() 
+        
+    @nextcord.ui.button(label='Nah, I am sorry!', style=nextcord.ButtonStyle.danger)
+    async def Cancel(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
+        await interaction.response.send_message('Cancelled', ephemeral=True)
+        global help_topic
+        help_topic = 'None'
```
Comment (sairam4123) [inline cogs/help.py:51]: should not be global

**Review by sairam4123 — CHANGES_REQUESTED** [review]

File: cogs/help.py (hunk 2 above)
Comment (AakashS-chan) [reply cogs/help.py:42]: It was hardcoded in the original code also, i took these things from there to keep it same and simple

File: cogs/help.py [hunk 4]
```diff
@@ -3,29 +3,52 @@
...
+   
+class Confirmation(nextcord.ui.View): 
+    
+    def __init__(self, channel_name):
+        super().__init__()
+        self.name = channel_name       
+        
+    @nextcord.ui.button(label='Yes!', style=nextcord.ButtonStyle.success)
+    async def confirm(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
+        await self.create_help_thread(name=self.name, interaction=interaction)
```
Comment (sairam4123) [inline cogs/help.py:34]: Don't create the thread with the python or those kinds of stuff, I'd recommend doing this.

First, let the user select the button, then ask him what he needs help for, and then send it and add people with pings.

Comment (AakashS-chan) [reply cogs/help.py:34]: "what he needs help for, and then send it and add people with pings." What do you mean by then send it? BTW I just added the confirm buttons and changed what was necessary to add the buttons, the rest is the same as the previous code.

File: cogs/help.py [hunk 5]
```diff
@@ -3,29 +3,52 @@
...
-        await self.create_help_thread("Nextcord", interaction)
-    @ui.button(label="D.py libs help", style=ButtonStyle.gray, custom_id="help:dpylibs")
-    async def nextcord_lib_help(self, _, interaction: Interaction):
-        await self.create_help_thread("Discord.py libraries", interaction)
+class HelpView(nextcord.ui.View):
+    
+    @nextcord.ui.button(label='Nextcord Help?', style=nextcord.ButtonStyle.green)
+    async def nextcord_(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
+        view = Confirmation('Nextcord')
+        await interaction.response.send_message('Click "Yes!" to create the thread.', view=view, ephemeral=True)
```
Comment (TAG-Epic) [inline cogs/help.py:11]: Possibly change it to something like "Are you really sure you want to make a help thread?"

File: cogs/help.py [hunk 6]
```diff
@@ -3,29 +3,52 @@
...
+        super().__init__()
+        self.name = channel_name       
+        
+    @nextcord.ui.button(label='Yes!', style=nextcord.ButtonStyle.success)
+    async def confirm(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
+        await self.create_help_thread(name=self.name, interaction=interaction)
+        await interaction.response.send_message('Created the thread', ephemeral=True)
+        self.stop() 
+        
+    @nextcord.ui.button(label='Nah, I am sorry!', style=nextcord.ButtonStyle.danger)
```
Comment (TAG-Epic) [inline cogs/help.py:38]: This should just be "Yes" and "No""

File: cogs/help.py [hunk 7]
```diff
@@ -3,29 +3,52 @@
...
+    async def cancel(self, button: nextcord.ui.Button, interaction: nextcord.Interaction):
+        await interaction.response.send_message('Cancelled', ephemeral=True)
```
Comment (TAG-Epic) [inline cogs/help.py:40]: If they press No there shouldnt really be a response to that except the buttons turning off imo

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

**Comment by AakashS-chan** [comment]
Feat

**Comment by AakashS-chan** [comment]
Feature already in a pr

**Pull request closed without merging**